cumulus/parachains/runtimes/assets/asset-hub-westend/src/xcm_config.rs

```diff
@@ -349,6 +349,27 @@ pub type TrustedTeleporters = (
 	IsForeignConcreteAsset<FromSiblingParachain<parachain_info::Pallet<Runtime>>>,
 );
 
+pub struct SystemAccountId32Aliases;
+impl ContainsPair<Location, Location> for SystemAccountId32Aliases {
+	fn contains(original: &Location, target: &Location) -> bool {
+		log::error!("original {:?}", original);
+		let original_id = match original.unpack() {
+			(1, [Parachain(1001), AccountId32 { network: None, id }]) => id,
+			(1, [Parachain(1001), AccountId32 { network, id }])
+				if *network == RelayNetwork::get() =>
+				id,
```

Comment:
I don't quite follow this. Does this mean it only recognizes origins from `Parachain(1001)` as the source? So each chain would have to configure all the chains it recognizes as source chains? Makes sense, just want to make sure I understand.

Reply:
Right,  relation where an origin of `location { parent: 1, [Parachain(%para_id%), AccountId32(%id%)]}` has it's alias as `location { parent: 0, [AccountId32(%id%)]}` is true only for authorised list of %para_id%s
This was implemented only for demonstration purpose, if we move forward with the solution, I probably move this to xcm builder and make it accept the list of para ids.